addons/sale/models/sale_order.py

```diff
@@ -820,7 +820,14 @@ def action_confirm(self):
         context.pop('default_name', None)
 
         self.with_context(context)._action_confirm()
-        if self.env.user.has_group('sale.group_auto_done_setting'):
+
+        # Before we had this condition checking self.env.user but this won't work
+        # if the SO is coming from eCommerce with public user env
+        # So now we are checking the creator of the SO that in the eCommerce
+        # scenario will be OdooBot and if the SO is created from the backend
+        # it will be one of the users who already has the group
+
+        if self.create_uid.has_group('sale.group_auto_done_setting'):
```

Comment:
The complete explanation should be inside the commit message, not the code. This should be sufficient as a comment:

```suggestion
        if self.create_uid.has_group('sale.group_auto_done_setting'):  # Public user can confirm SO
```